packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/nix/changeArgBlueDot.yml

```diff
@@ -0,0 +1,39 @@
+languageId: nix
+command:
+  version: 6
+  spokenForm: change arg blue dot
+  action:
+    name: clearAndSetSelection
+    target:
+      type: primitive
+      mark: {type: decoratedSymbol, symbolColor: blue, character: .}
+      modifiers:
+        - type: containingScope
+          scopeType: {type: argumentOrParameter}
+  usePrePhraseSnapshot: true
+initialState:
+  documentContents: |-
+    { lib, buildPackages, fetchurl, stdenv, ... }:
```

Comment:
what does the `...` do?

Reply:
It basically allows you to ignore/allow all non-explicitly defined attributes in the set. If you don't include it and someone passes a set to the function including any attribute not explicitly included, it will fail. So ... allows you to accept a set with unexpected attributes.

It actually gets a bit more complicated though, which I haven't figured it if/how I'll deal with yet. The set passed as an argument has special syntax like foo@{ a, b, ... } where after destructuring you can still refer to the set by the name foo, in which case you can then reference one of the variables swallowed by ... explicitly lol. I'm not sure if there is any scope in cursorless suited to that foo@ part or if I just have to make sure arg references still work when that is included.

Reply:
interested. tempted to have `arg` refer to the entire `foo@{ a, b, ... }`. Then you could still use `item` for `a`, `b`, etc

Reply:
Ya I agree. That reminds me `item` is broken for nix lists, which are ` `-delimited not `,`-delimited. `item` does work well for `,`-delimited sets though. Perhaps that's better discussed in a separate comment thread though. Basically though `change item` in a nix list will just change the entire contents of the list, not a single item, I guess because there is no `,` found.

Reply:
fwiw I added a file `complex.nix` to the `data/playground/nix/` to show the real-world complicated list usage that one will encounter when coding nix, and to just demonstrate that `item` doesn't currently work. See the `modules = [` on line 16 for example.

This file also has some nice adversarial lambda usage that highlights some weaknesses in the queries I have atm.

Reply:
If you define the `collectionItem` scope in places where it's space delimiter, it should use that there, and fallback to comma delimiter elsewhere